tf_encrypted/protocol/pond.py

```diff
@@ -3526,6 +3556,77 @@ def _reshape_masked(
             x.is_scaled,
         )
 
+#
+# neg helpers
+#
+
+
+def _neg_public(
+    prot: Pond, x: PondPublicTensor
+) -> PondPublicTensor:
+    assert isinstance(x, PondPublicTensor)
+
+    x_on_0, x_on_1 = x.unwrapped
+
+    with tf.name_scope("neg"):
+
+        with tf.device(prot.server_0.device_name):
+            x_on_0_neg = x_on_0.neg()
+
+        with tf.device(prot.server_1.device_name):
+            x_on_1_neg = x_on_1.neg()
+
+        return PondPublicTensor(prot, x_on_0_neg, x_on_1_neg, x.is_scaled)
+
+
+def _neg_private(
+    prot: Pond, x: PondPrivateTensor
+) -> PondPrivateTensor:
+    assert isinstance(x, PondPrivateTensor)
+
+    x0, x1 = x.unwrapped
+
+    with tf.name_scope("neg"):
+
+        with tf.device(prot.server_0.device_name):
+            x0_neg = x0.neg()
+
+        with tf.device(prot.server_1.device_name):
+            x1_neg = x1.neg()
+
+        return PondPrivateTensor(prot, x0_neg, x1_neg, x.is_scaled)
+
+
+def _neg_masked(
```

Comment:
this works out but just a small note of caution that this is not always straight-forward and could lead to privacy issues 👍